scheduler/replica_scheduler.go

```diff
@@ -320,6 +321,13 @@ func (rcs *ReplicaScheduler) filterNodeDisksForReplica(node *longhorn.Node, disk
 			continue
 		}
 
+		volumeSize := volume.Spec.Size
+		// unix.Statfs can not differentiate the ext2/ext3/ext4 file systems.
+		if (strings.HasPrefix(string(diskStatus.Type), "ext") && volumeSize >= util.MaxExt4VolumeSize) || (string(diskStatus.Type) == "xfs" && volumeSize >= util.MaxXfsVolumeSize) {
+			logrus.Debugf("Volume %v size %v is not compatible with the file system %v of the disk %v", volume.Name, volume.Spec.Size, diskStatus.Type, diskName)
+			continue
+		}
+
```

Comment:
volume validator already rejects volumes with the conditions. Do you need to add the check here?

Reply:
The volume validator will not reject that users create a volume over 16TiB because we don't know if there are disks that are formatted to "ext4", "xfs" or others until we create the volume and it starts to schedule the replicas to disks.

@derekbit can you point out where the volume validator rejects creating volumes?

Reply:
I think we do. We should not prevent users from creating a 16TiB volume in case there are at least some XFS replicas in the cluster. So we need to decide during replica scheduling which disks we can use.

Reply:
Got it. Thanks @mantissahz and @ejweber